src/lib.rs

```diff
@@ -120,23 +123,30 @@ impl RustNotify {
                             }
                         }
                         EventKind::Modify(ModifyKind::Name(RenameMode::From)) => CHANGE_DELETED,
-                        EventKind::Modify(ModifyKind::Name(RenameMode::To)) => CHANGE_ADDED,
-                        // RenameMode::Both duplicates RenameMode::From & RenameMode::To
-                        EventKind::Modify(ModifyKind::Name(RenameMode::Both)) => return,
-                        EventKind::Modify(ModifyKind::Name(_)) => {
-                            // On macOS the modify name event is triggered when a file is renamed,
-                            // but no information about whether it's the src or dst path is available.
-                            // Hence we have to check if the file exists instead.
-                            if Path::new(&path).exists() {
-                                CHANGE_ADDED
-                            } else {
-                                CHANGE_DELETED
-                            }
+                        EventKind::Modify(ModifyKind::Name(RenameMode::Both)) => {
+                            let mut changes = changes_clone.lock().unwrap();
+                            changes.remove(&(CHANGE_DELETED, path.clone(), EMPTY_STRING));
```

Comment:
Does this actually happen? If so, I guess we should remove `added` here too?

Reply:
It happens when a file is moved out of the directory. In this case we want to see it as deleted, which is handled in the `RenameMode::From` event. But if the file is moved in the same directory, we want to see it as moved, so we need to remove the `CHANGE_DELETED`. I didn't find a better way to do it.
When a file is moved in the directory from outside, we want to see it as added, this works fine already.

Reply:
makes sense.